### PR TITLE
Fixup/request builder

### PR DESCRIPTION
**Pull request by malakai97**
Multiple fixes/changes to request_builder. I was able to get as far into the client-driven process as you have previously (external_service).

### CODE REVIEW

File: lib/request_builder.rb [hunk 1]
```diff
@@ -19,6 +20,8 @@ def initialize(bag_id:, content_type:, external_id:, user:)
   end
 
   def create
+    Rails.logger.debug "making directory #{request.upload_path}"
+    fs.mkdir_p request.upload_path
```
Comment (aelkiss) [inline lib/request_builder.rb]: Unnecessary - the expectation is that the client has permissions to create this path

**Comment by aelkiss** [comment]
We should be more explicit about what the expectations for RequestBuilder/QueueItemBuilder are given that there's some reversion back and forth going on here. I think what you have here is fine but we should think carefully about naming and expectations about what's going on here. I had been thinking of those more as factories but I think you're thinking of them more as kind of an auxiliary or way of separating concerns for the controller, which makes sense given where it's called from, but I'm not sure what to name the classes and the methods that might make it clearer

**Comment by malakai97** [comment]
upload_path already contained the username in its path. This change was to make the upload directory match that. Without this change, running the end-to-end test fails because the uploaded files are not where they're expected to be.

RequestBuilder and QueueItemBuilder represent the use case.  The user creates a request. The user enqueues a task. That functionality should work outside of the request-response cycle. 

I create the upload directory because the rsync command fails without it.  The alternative is modifying the client to create the directory. This seemed cleaner.

**Comment by aelkiss** [comment]
1) end-to-end test fails: Here are my setups in dev as well as the 'real' setup in production. The key point here is that the client is not expected to upload to the same path that the server ends up retrieving the content from - these things should point to the same underlying storage but it isn't the same path.

`config/upload.yml`:
```yaml
development:
  upload_path: /tmp/pfdr/incoming
  rsync_point: localhost:incoming
  storage_path: /tmp/pfdr/storage
production:
  upload_path: /pfdr/incoming
  rsync_point: vodka.umdl.umich.edu:incoming
  storage_path: /pfdr/storage
```

Paths on the filesystem in dev:
* /tmp/pfdr/incoming/aelkiss
* /home/aelkiss/incoming -> symlink to /tmp/pfdr/incoming/aelkiss
* /tmp/pfdr/storage

Production:
* /home/aelkiss/incoming: CIFS mount of //ulib-dark-blue.m.storage.umich.edu/ulib-dark-blue/incoming/aelkiss with appropriate permissions so aelkiss can write to it. rsync/ssh is chrooted to /home/aelkiss so rsync/ssh sees /home/aelkiss as its root and can only CD into /incoming - this arrangement allows individual users to log in and upload with ssh keys to the Dark Blue CIFS mount while still enforcing permissions appropriately. (It could be simpler if it was on NFS or on a CIFS server with Unix extensions but neither of those are options because MiStorage.)
* /pfdr: CIFS mount of //ulib-dark-blue.m.storage.umich.edu/ulib-dark-blue so application can write to it
** /pfdr/incoming/aelkiss
** /pfdr/storage 

2) It might be helpful to see an example of how you might envision RequestBuilder becoming more complicated. Right now, you create a RequestBuilder with some parameters and then immediately call create on it. What kind of other things might you envision doing there? Not necessarily a question we need to answer right now, but something to keep in mind when things do get added in RequestBuilder and QueueItemBuilder to make sure we aren't going off in different directions.

3) Can you give a transcript of a CLI session where this fails?

**Pull request closed without merging**